src/common/xo/choose-sr-for-each-vdis-modal/index.js

```diff
@@ -0,0 +1,149 @@
+import Collapse from 'collapse'
+import Component from 'base-component'
+import every from 'lodash/every'
+import forEach from 'lodash/forEach'
+import map from 'lodash/map'
+import React from 'react'
+
+import _ from '../../intl'
+import SingleLineRow from '../../single-line-row'
+import { createSelector } from '../../selectors'
+import {
+  Container,
+  Col
+} from 'grid'
+import {
+  SelectSr
+} from '../../select-objects'
+import {
+  isSrWritable
+} from 'xo'
+
+// Can 2 SRs on the same pool have 2 VDIs used by the same VM
+const areSrsCompatible = (sr1, sr2) =>
+  sr1.shared || sr2.shared || sr1.$container === sr2.$container
+
+class _Collapsible extends Component {
+  render () {
+    const {
+      collapsible,
+      children,
+      ...props
+    } = this.props
+
+    return collapsible
+      ? <Collapse {...props}>{children}</Collapse>
+      : <div>
+        <span>{props.buttonText}</span>
+        <br />
+        {children}
+      </div>
+  }
+}
+
+export default class ChooseSrForEachVdisModal extends Component {
+  constructor () {
+    super()
+
+    this.state = {
+      mapVdisSrs: {}
+    }
+  }
+
+  componentWillReceiveProps (newProps) {
+    if (
+      this.props.predicate !== undefined &&
+      newProps.predicate !== this.props.predicate
+    ) {
+      this.state = {
+        mainSr: undefined,
+        mapVdisSrs: {}
+      }
+    }
+  }
+
+  _onChange = props => {
+    this.setState(props)
+    this.props.onChange(props)
+  }
```

Comment:
It's weird to both call a `onChange()` function which suggest that data is stored in the props and store them in the state.